src/transformStateWithClones.js

```diff
@@ -65,7 +65,27 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const result = [];
+
+  const stateClone = { ...state };
+
+  for (const transform of transforms) {
+    if (transform.operation === 'addProperties') {
+      Object.assign(stateClone, transform.properties);
+      result.push({ ...stateClone });
```

Comment:
do push only once at the end of the loop

Reply:
done